src/lib/services/access-service.ts

```diff
@@ -662,32 +679,63 @@ export class AccessService {
         }
 
         await this.validateRole(role, role.id);
+        const existingRole = await this.roleStore.get(role.id);
         const baseRole = {
             id: role.id,
             name: role.name,
             description: role.description,
             roleType,
         };
         const rolePermissions = role.permissions;
-        const newRole = await this.roleStore.update(baseRole);
+        const updatedRole = await this.roleStore.update(baseRole);
+        const existingPermissions = await this.store.getPermissionsForRole(
+            role.id,
+        );
         if (rolePermissions) {
-            await this.store.wipePermissionsFromRole(newRole.id);
+            await this.store.wipePermissionsFromRole(updatedRole.id);
             if (roleType === CUSTOM_ROOT_ROLE_TYPE) {
                 await this.store.addPermissionsToRole(
-                    newRole.id,
+                    updatedRole.id,
                     rolePermissions,
                 );
             } else {
                 await this.store.addEnvironmentPermissionsToRole(
-                    newRole.id,
+                    updatedRole.id,
                     rolePermissions,
                 );
             }
         }
-        return newRole;
+        const updatedPermissions = await this.store.getPermissionsForRole(
+            role.id,
+        );
+        this.eventService.storeEvent({
+            type: ROLE_UPDATED,
+            createdBy: role.createdBy || 'unknown',
+            data: {
+                ...updatedRole,
+                permissions: this.sanitizePermissions(updatedPermissions),
+            },
+            preData: {
+                ...existingRole,
+                permissions: this.sanitizePermissions(existingPermissions),
+            },
+        });
+        return updatedRole;
+    }
+
+    sanitizePermissions(
```

Comment:
where do we test sanitization logic?

Reply:
https://github.com/Unleash/unleash/pull/5408/files/c1cdc0efb24a500f14e5b64d45f906369e11e818#diff-a6135f07806e2e7dbc233652abfbd18fe122690638252d488eb6f82ae8f02e10R198-R212 covers the sanitization logic. But can have specific UT